### DISCUSSION TITLE
Providing an upper and lower bound of the parameters in the fitting?

### DISCUSSION THREAD

**Discussion by mubali101** [georgebv/pyextremes#75] [Q&A]
Thanks for creating and maintaining this package. 

While fitting a model using MLE, one can fix a certain parameter. Is it possible to give an upper and lower bound of the parameter instead of one fixed value? 
Example something around the line below, where I would like the MLE to pick best value of `loc` in the bounds 270 to 272
````
model.fit_model(distribution="genpareto", distribution_kwargs=dict(floc=[270,272]))

````

**Comment by georgebv** [comment]
These kwargs come from [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.fit.html) which doesn't support this.

With `genpareto`, however, you **must** know you location parameter precisely - it's your threshold, by definition.

**Reply by mubali101** [reply]
Thanks for the quick reply. I guess the workaround would be to write a custom fit function using `scipy.optimize.minimize()` as it supports giving bounds. 

Another small point I would like to confirm when comparing the results with scipy.fit() directly is that I'm getting differences in the second digit after decimal in the shape and scale parameters. I think that should be acceptable but still curious as to why such a difference can be if `pyextremes` is internally using `scipy.fit()` function as well. 

I have copied the code below in case if there is some issue with the way I'm fitting. I think when using scipy.fit() directly, in case of GPD with extremes obtained by POT, after fixing the location parameter, I think one doesn't need to provide exceedances of the data for fitting as the distribution is shifted to that fixed location. Happy to share the time series if you would like to reproduce the results. 
 
``` python
model = EVA(data=data)
model.get_extremes( threshold=thresh,
method="POT", extremes_type="low",
 r="24H", )
model.fit_model(distribution="genpareto", distribution_kwargs=dict(floc=thresh))

# get extremes
df_extremes = model.extremes

def transform_data(data, pivot_point=thresh):
    """Since GPD is fitted to the lower tail in pyextremes 
    by transforming the data along the pivot point    
    """
    return 2*pivot_point - data

data_transformed = transform_data(df_extremes.values)

# Fit GPD to df_extremes using scipy
floc = thresh # fix location 
params = stats.genpareto.fit(data_transformed,floc=floc, method='MLE')

print(params) # shape, loc and scale
>> (-0.10268956690420433, 272.051884765625, 2.5898906079681496)

print(model.model.distribution)
>>      pyextremes distribution       
-------------------------------------
name: genpareto                      
free parameters: c, scale            
fixed parameters: floc=272.052       
MLE parameters: c=-0.092, scale=2.538
-------------------------------------


```

**Reply by georgebv** [reply]
Try setting `floc` to the min (or max) of your extremes. That's what's used as your pivot point and your floc

**Reply by mubali101** [reply]
Thanks @georgebv. That solved the issue and the results match with scipy.fit() now.
